tests/system/feature_store/test_feature_store.py

```diff
@@ -968,6 +968,192 @@ def test_filtering_parquet_by_time(self, with_tz, local):
         with pytest.raises(FileNotFoundError):
             result_offline_target.as_df()
 
+    @TestMLRunSystem.skip_test_if_env_not_configured
+    @pytest.mark.parametrize(
+        "partition_keys",
+        [
+            ["year"],
+            ["year", "month"],
+            ["year", "month", "day"],
+            ["year", "month", "day", "hour"],
+        ],
+    )
+    @pytest.mark.parametrize("with_tz", [True, False])
+    def test_partitioned_parquet_as_df(self, partition_keys, with_tz):
```

Comment:
```suggestion
    def test_partitioned_parquet_as_df_time_filtering_optimization(self, partition_keys, with_tz):
```
We're not just testing `as_df` in general here.